src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -58,9 +58,27 @@ internal QueryNode BindInOperator(InToken inToken, BindingState state)
                     inToken.Right, new EdmCollectionTypeReference(new EdmCollectionType(EdmCoreModel.Instance.GetUntyped())), state.Model);
             }
 
+            if (ShouldConvertLeftOperand(left, right))
+            {
+                left = MetadataBindingUtils.ConvertToTypeIfNeeded(left, right.ItemType);
+            }
+
             return new InNode(left, right);
         }
 
+        /// <summary>
+        /// Checks if the left operand is either an integral or a string type and the right operand is a collection of enums.
+        /// </summary>
+        /// <param name="leftNode">The left operand.</param>
+        /// <param name="rightNode">The right operand.</param>
+        /// <returns>True if the condition is met, otherwise false.</returns>
+        private static bool ShouldConvertLeftOperand(SingleValueNode leftNode, CollectionNode rightNode)

```

Comment:
We currently have a method that promotes types to compatible types, why can't we use it instead of adding this method to do this check? Can the check be added to that method if need be?

Reply:
@ElizabethOkerio This is just a private method that checks for the condition before calling `MetadataBindingUtils.ConvertToTypeIfNeeded` which is the existing method that does the promotion. It was for readability purposes. 

```cs
if (ShouldConvertLeftOperand(left, right))
{
    left = MetadataBindingUtils.ConvertToTypeIfNeeded(left, right.ItemType);
}
```

where:
```cs
private static bool ShouldConvertLeftOperand(SingleValueNode leftNode, CollectionNode rightNode)
{
    // If the left operand is either an integral or a string type and the right operand is a collection of enums,
    // Calls the MetadataBindingUtils.ConvertToTypeIfNeeded() method to convert the left operand to the same enum type as the right operand.
    return (rightNode is CollectionPropertyAccessNode && rightNode.ItemType.IsEnum()) && (leftNode.TypeReference.IsString() || leftNode.TypeReference.IsIntegral());
}
```

If not necessary, I can remove the ` bool ShouldConvertLeftOperand(SingleValueNode leftNode, CollectionNode rightNode)' and put the condition as below:
```cs
if ((rightNode is CollectionPropertyAccessNode && rightNode.ItemType.IsEnum()) && (leftNode.TypeReference.IsString() || leftNode.TypeReference.IsIntegral()))
{
    left = MetadataBindingUtils.ConvertToTypeIfNeeded(left, right.ItemType);
}
```